packages/solid-loaders/src/index.tsx

```diff
@@ -0,0 +1,164 @@
+import {
+  Loader,
+  LoaderClient,
+  LoaderClientOptions,
+  LoaderClientStore,
+  LoaderInstance,
+  LoaderInstanceByKey,
+  LoaderStore,
+  RegisteredLoaders,
+  StrictLoaderInstance,
+  VariablesOptions,
+} from '@tanstack/loaders'
+import * as Solid from 'solid-js'
+import * as SolidStore from 'solid-js/store'
+import invariant from 'tiny-invariant'
+
+import { useStore } from '@tanstack/solid-store'
+
+export * from '@tanstack/loaders'
+
+//
+
+export type NoInfer<T> = [T][T extends any ? 0 : never]
+
+const loaderClientContext = Solid.createContext<LoaderClient<any>>(null as any)
+
+export function LoaderClientProvider(
+  props: {
+    loaderClient: LoaderClient<any>
+    children: any
+  } & Pick<LoaderClientOptions<any>, 'defaultGcMaxAge' | 'defaultMaxAge'>,
+) {
+  const [coreProps, optionProps] = Solid.splitProps(props, [
+    'loaderClient',
+    'children',
+  ])
+
+  const [loaderClient, setLoaderClient] = SolidStore.createStore(
+    coreProps.loaderClient,
+  )
+
+  Solid.createEffect(
+    Solid.on(
+      () => [coreProps.loaderClient.options, optionProps],
+      ([loaderOptions, optionProps]) => {
+        setLoaderClient('options', (o) => ({
+          ...o,
+          ...loaderOptions,
+          ...optionProps,
+        }))
+      },
+    ),
+  )
+
+  return (
+    <loaderClientContext.Provider value={loaderClient}>
+      {coreProps.children}
+    </loaderClientContext.Provider>
+  )
+}
+
+type StrictLoaderInstance_<TLoaderInstance> =
+  TLoaderInstance extends LoaderInstance<
+    infer TKey,
+    infer TVariables,
+    infer TData,
+    infer TError
+  >
+    ? StrictLoaderInstance<TKey, TVariables, TData, TError>
+    : never
+
+export function useLoaderInstance<
+  TKey extends RegisteredLoaders[number]['__types']['key'],
+  TLoader,
+  TStrict,
+  TLoaderInstanceFromKey extends LoaderInstanceByKey<RegisteredLoaders, TKey>,
+  TResolvedLoaderInstance extends unknown extends TLoader
+    ? TLoaderInstanceFromKey
+    : TLoader extends Loader<
+        infer TTKey,
+        infer TTVariables,
+        infer TTData,
+        infer TTError
+      >
+    ? LoaderInstance<TTKey, TTVariables, TTData, TTError>
+    : never,
+  TVariables extends TResolvedLoaderInstance['__types']['variables'],
+  TData extends TResolvedLoaderInstance['__types']['data'],
+  TError extends TResolvedLoaderInstance['__types']['error'],
+>(
+  opts: (
+    | {
+        key: TKey
+      }
+    | {
+        loader: TLoader
+      }
+  ) & {
+    strict?: TStrict
+    track?: (loaderStore: LoaderStore<TData, TError>) => any
+    throwOnError?: boolean
+  } & VariablesOptions<TVariables>,
+): TStrict extends false
+  ? TResolvedLoaderInstance
+  : StrictLoaderInstance_<TResolvedLoaderInstance> {
```

Comment:
This condition need to be swapped for the change to `NullableLoaderInstance` instead of strict